### ISSUE TITLE
`ComboBox`'s `Flyout` does not respect visual transform

### ISSUE THREAD

**Issue by pr8x** [AvaloniaUI/Avalonia#7147]
![image](https://user-images.githubusercontent.com/4670166/145859705-69bf322a-e5c1-4bfb-978a-07a03d1eb19f.png)

Pretty self-explanatory. The `ComboBox` has some `ScaleTransform` applied but its `Flyout`/`Popup` does not respect that.

**Comment by maxkatz6** [comment]
Flyout or just combobox dropdown menu?
Because I don't think scale transform should apply on Flyout/ContextMenu.
But it should be applied on combobox drop down menu.

**Comment by pr8x** [comment]
Yeah I mean the Dropdown menu (I assume it was implemented in terms of flyouts)

**Comment by grokys** [comment]
Just checked with WPF and indeed it applies the render transform to the drop-down menu. It even works with rotate transforms, which surprised me!

![image](https://user-images.githubusercontent.com/1775141/146163211-26f3cc5d-ab9e-4ea8-b56a-79c170f2a69f.png)
